apps/explorer-e2e/src/support/pages/transactions-page.js

```diff
@@ -48,35 +43,51 @@ export default class TransactionsPage extends BasePage {
   }
 
   validateTxDetailsAreDisplayed() {
-    cy.getByTestId(this.txHash).invoke('text').should('have.length', 64);
-    cy.getByTestId(this.txSubmittedBy)
-      .find('a')
-      .then(($address) => {
-        cy.wrap($address).should('have.attr', 'href');
-        cy.wrap($address).invoke('text').should('have.length', 66);
-      });
-    cy.getByTestId(this.txBlock).should('not.be.empty');
-    cy.getByTestId(this.txEncodedTnx).should('not.be.empty');
-    cy.getByTestId(this.txType)
-      .should('not.be.empty')
-      .invoke('text')
-      .then((txTypeTxt) => {
-        if (txTypeTxt == 'Order Submission') {
-          cy.get('.hljs-attr')
-            .should('have.length.at.least', 8)
-            .each(($propertyName) => {
-              cy.wrap($propertyName).should('not.be.empty');
-            });
-          cy.get('span[style*="color"]')
-            .should('have.length.at.least', 8)
-            .each(($propertyValue) => {
-              cy.wrap($propertyValue).should('not.be.empty');
-            });
-        }
-      });
+    cy.wait(1000);
+    cy.get('body').then(($body) => {
+      if ($body.find(`[data-testid=${this.txHash}]`).length) {
+        cy.getByTestId(this.txHash).invoke('text').should('have.length', 64);
+        cy.getByTestId(this.txSubmittedBy)
+          .find('a')
+          .then(($address) => {
+            cy.wrap($address).should('have.attr', 'href');
+            cy.wrap($address).invoke('text').should('have.length', 66);
+          });
+        cy.getByTestId(this.txBlock).should('not.be.empty');
+        cy.getByTestId(this.txEncodedTnx).should('not.be.empty');
+        cy.getByTestId(this.txType)
+          .should('not.be.empty')
+          .invoke('text')
+          .then((txTypeTxt) => {
+            if (txTypeTxt == 'Order Submission') {
+              cy.get('.hljs-attr')
+                .should('have.length.at.least', 8)
+                .each(($propertyName) => {
+                  cy.wrap($propertyName).should('not.be.empty');
+                });
+              cy.get('span[style*="color"]')
```

Comment:
I'm a bit dubious of targeting an element with a style property, I expect this could break easily on a change to the UI theme designs

Reply:
I totally agree finding element by style is not best practice. A while ago I did see if I could add test ids to the json data pulled in for the tx details but I don't think it was as easy as the other elements. I'll create a separate issue for that.